### DISCUSSION TITLE
Importing Prisma Client Models in Bun

### DISCUSSION THREAD

**Discussion by Pola-Malak** [oven-sh/bun#14087] [Q&A]
Hi everyone!

I am new to Bun and experimenting with migrating an existing NestJS Backend node app to Bun. However, I’m encountering issues with importing models from Prisma Client while working with Bun v1.1.29.

**Problem**: Named imports throw errors, like:

`import { User } from '@prisma/client';`  

throws something like this even thought it works with Node:

`SyntaxError: Export named 'User' not found in module '/path/to/node_modules/@prisma/client/default.js'.`

This issue arises when I try to import any other Prisma models as well. These imports work perfectly with Node.js but fail in Bun.

**Code Example:**
Here's an example userService.ts file that shows how I’m trying to import and use Prisma models in my project:

```ts
import { PrismaService } from 'path/to/prisma.service';
import { User } from '@prisma/client'

@Injectable()
export class UserService {

  constructor() {
    private readonly prismaService: PrismaService,
  }

  async getUserById(userId: number): User {
    try {
      const user = await this.prismaService.user.findUnique({
        where: { id: userId },
      });
      return user;
    } catch (error) {
      console.error('Error fetching user:', error);
      throw error;
    }
  }
}
```
PrismaService.ts
```ts
import { Injectable, OnModuleInit } from '@nestjs/common';
import { PrismaClient } from '@prisma/client';

@Injectable()
export class PrismaService extends PrismaClient implements OnModuleInit {
  async onModuleInit() {
    await this.$connect();
  }
}
```
Similarly:

`import { Book,Author } from '@prisma/client';  // Throws error: 'Export named Book not found in module path/to/node_modules/@prisma/client/default.js ' `

This seems to relate to how Bun handles ESM and CommonJS modules. Bun doesn't recognize the named exports, although Prisma clearly exports them as types in the index.js file.

From the generated Prisma Client files, I can see the models are defined as types.

**Setup:**
Bun version: v1.1.29
Prisma version: v5.18.0
Node.js version: Works perfectly with Node 20.x

**tsconfig.json used:**

```json
{
  "compilerOptions": {
    "module": "commonjs",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "target": "ES2021",
    "sourceMap": true,
    "outDir": "./dist",
    "baseUrl": "./",
    "incremental": true,
    "skipLibCheck": true,
    "strictNullChecks": false,
    "noImplicitAny": false,
    "strictBindCallApply": false,
    "forceConsistentCasingInFileNames": false,
    "noFallthroughCasesInSwitch": false,
    "resolveJsonModule": true
  }
}
// Also used tsconfig suggested By Bun in Docs but did not work
```

**Questions:**

- How can I get Bun to handle named imports from Prisma Client properly?

- Is there a specific tsconfig.json or Bun configuration I need for TypeScript compatibility?

Thank you for your help!

**Comment by Renato66** [comment]
did you tried prisma generate?

**Reply by Pola-Malak** [reply]
yes sure and all the models are already showing in Prisma index.js

**Comment by czernika** [comment]


**Reply by kravetsone** [reply]
> Same issue here, bun v1.1.43,
> 
> But I've found you can import it under namespace, e.g
> 
> ```js
> import Prisma from '@prisma/client';
> 
> @Get('/me')
> async me(@CurrentUser() user: Prisma.User) { // Here, no error
>   return this.usersService.findOne({
>     id: user.id,
>   })
> }
> ```
> 
> in your case `Prisma.Book`, `Prisma.Author`, etc. Instead of `Prisma` it can be anything meaningful to you

why not using `import type`?

**Reply by czernika** [reply]
Didn't even thought it would help. Thanks!
